### DISCUSSION TITLE
Change color of image pixel

### DISCUSSION THREAD

**Discussion by Ron5s** [Kode/Kha#1386] [General] [answered]
Is there a way to set the pixel color of an image other than shaders?

Thanks for the help.

**Comment by RobDangerous** [comment, marked as answer]
You can use lock/unlock to change an image but you have to write all the pixels and don't get access to their old values that way. You can access the original data by calling Image.at when the image is set to readable (see https://github.com/Kha-Samples/TPlayer/blob/master/khafile.js for how to do that for images that are handled by the Assets class). But all of that is highly discouraged because it is super-slow, use shaders instead if you can.

**Reply by Ron5s** [reply]
Understood.  thank you for the help
